catalog/db_diff_scanner.go

```diff
@@ -86,12 +85,31 @@ func (s *DiffScanner) diffFromParent(tx db.Tx, params doDiffParams) (*DiffScanne
 		After:            params.After,
 		AdditionalFields: prepareDiffAdditionalFields(params.AdditionalFields),
 	}
-	s.leftScanner = NewDBLineageScanner(tx, params.LeftBranchID, CommittedID, scannerOpts)
-	s.rightScanner = NewDBLineageScanner(tx, params.RightBranchID, UncommittedID, scannerOpts)
-	s.childLineage, err = getLineage(tx, params.RightBranchID, UncommittedID)
+	parentLineage, err := getLineage(tx, params.LeftBranchID, CommittedID)
 	if err != nil {
 		return nil, err
 	}
+	childLineage, err := getLineage(tx, params.RightBranchID, UncommittedID)
+	if err != nil {
+		return nil, err
+	}
+	// If some ancestor branch commit id is the same for parent and child - then the parent does not need to read it
+	// so it is trimmed from the parent lineage
+	if len(parentLineage) >= 1 {
```

Comment:
1. no need to check len >= 1 if it was 0 the loop was skipped
2. need to verify that len(right lineage) > len(left lineage)

Reply:
done